### PR TITLE
Feature/add national service option

### PR DESCRIPTION
**Pull request by appsol**
### Add in a new "national" service option

Ref: https://trello.com/c/fDs5TiSH

- Added Migration for Boolean is_national field. Defaults to true.
- Updated Service model and factory to reflect change
- Added is_national to Service validation for create and update
- Create Rule IsLocalService to check the Service is_national flag.
- Updated the ServiceLocation validation to prevent allocating locations to national services
- Added is_national to search parameters for Services
- Created tests to check for filter by both is_national == true and is_national == false

### CODE REVIEW

**Review by matthew-inamdar — CHANGES_REQUESTED** [review]
Really good stuff! Left a comment regarding indexing the new column, but apart from that, this looks really solid 💪

File: database/migrations/2020_09_22_131554_add_is_national_field_to_services.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+class AddIsNationalFieldToServices extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up()
+    {
+        Schema::table('services', function (Blueprint $table) {
+            $table->boolean('is_national')->after('status')->default(true);
```
[the remark this answers is not in the document]
Comment (appsol) [reply database/migrations/2020_09_22_131554_add_is_national_field_to_services.php:15]: I'm happy to do it if you think it's worthwhile, but I'm not sure the gains from indexing a boolean field will be that much.

[the remark this answers is not in the document]
Comment (matthew-inamdar) [reply database/migrations/2020_09_22_131554_add_is_national_field_to_services.php:15]: It's worth putting it in now as it doesn't take much time. It's one less thing to worry about if we need to scale this up to support many more services.

[the remark this answers is not in the document]
Comment (appsol) [reply database/migrations/2020_09_22_131554_add_is_national_field_to_services.php:15]: Done.

**Review by matthew-inamdar — DISMISSED** [review]

**Comment by matthew-inamdar** [comment]
Waiting to merge until https://github.com/Healthy-London-Partnership/admin/pull/13 is also ready.

**Review by matthew-inamdar — CHANGES_REQUESTED** [review]
The code looks great 👍 

When manual testing, I noticed you can have an existing non-national service with a service location, and then update it to become national.

We should prevent this from happening with request validation.

**Comment by appsol** [comment]
Requested changes completed.

**Review by matthew-inamdar — APPROVED** [review]

**Pull request merged**